app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,53 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find(params[:id])
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+    @task.save
+    if @task.save
+      redirect_to tasks_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @task = Task.find(params[:id])#gives you edit form, like new for create
+  end
+
+  def update #like create, but for editing
+    @task = Task.find(params[:id])
+    if @task.update(task_params)
+      redirect_to tasks_path
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @task = Task.find(params[:id]).destroy
+    redirect_to tasks_path
+  end
+
+  def record_completion
+    @task = Task.find(params[:id][:completion]).Datetime.now
```

Comment:
Let's break down each piece of this line to see where we're going wrong:
- What _should_ `Task.find` give you? 
- By adding the `.Datetime.now` to the result of `find`, what will that do?
- In other examples that we have, what is the parameter that we pass to `find`?
- In this example, what is the parameter we are passing to find?

Overall, you have the right idea, but the syntax is just a bit off.
```
@task = Task.find(params[:id]) # Find the task
@task.completion = DateTime.now # Update the task
@task.save # Save the updates
```